### PR TITLE
Add missing TryFrom impls for holo hash conversions

### PR DESCRIPTION
**Pull request by maackle**
### Summary

See changelog

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/holo_hash/src/aliases.rs [hunk 1]
```diff
@@ -81,6 +83,24 @@ impl AnyLinkableHash {
...
+    /// Downcast to AnyDhtHash if this is not an external hash
+    pub fn into_any_dht_hash(self) -> Option<AnyDhtHash> {
+        match self.into_primitive() {
+            AnyLinkableHashPrimitive::Action(hash) => Some(AnyDhtHash::from(hash)),
+            AnyLinkableHashPrimitive::Entry(hash) => Some(AnyDhtHash::from(hash)),
+            AnyLinkableHashPrimitive::External(_) => None,
+        }
+    }
+
+    /// If this hash represents a ActionHash, return it, else None
```
Comment (jost-s) [inline crates/holo_hash/src/aliases.rs:95]:
```suggestion
    /// If this hash represents an ActionHash, return it, else None
```

File: crates/holo_hash/src/aliases.rs [hunk 2]
```diff
@@ -120,6 +144,15 @@ impl AnyDhtHash {
         }
     }
 
+    /// If this hash represents a ActionHash, return it, else None
```
Comment (jost-s) [inline crates/holo_hash/src/aliases.rs:147]:
```suggestion
    /// If this hash represents an ActionHash, return it, else None
```

**Review by jost-s — APPROVED** [review]
LGTM

**Review by neonphog — APPROVED** [review]
Seems like it could be a good opportunity for a macro to DRY it up a bit, but not required. Looks good.

**Pull request merged**